toolkit/tools/pkg/imagecustomizerlib/kernelmoduleutils_test.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnsureModulesLoaded(t *testing.T) {
+	if os.Geteuid() != 0 {
+		t.Skip("Test must be run as root because it uses a chroot")
+	}
+
+	// Setup environment.
+	proposedDir := filepath.Join(tmpDir, "TestEnsureModulesLoaded")
+	chroot := safechroot.NewChroot(proposedDir, false)
```

Comment:
Is `chroot` actually needed here? It seems like you are just reading/writing files.